### PR TITLE
Fix Command+A command on Mac

### PR DESCRIPTION
**Pull request by sangwoo108**
If we pass 0 as keycode of NSEvent, it could be translated as
'a' key. So we should pass the actual keycode.

Resolves https://github.com/brave/brave-browser/issues/31059

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
* Open up multiple tabs
* Focus omnibox and type some text. Make sure you don't select all text already.
* Press Cmd + A or activate app menu > edit > select all
* Should select all texts.

### CODE REVIEW

**Comment by sangwoo108** [comment]
@simonhong , I've checked on my machine but could you double check if this works?

**Review by fallaciousreasoning — APPROVED** [review]
Wow, gnarly issue @sangwoo108! Good find, I bet that wasn't easy to track down

**Review by simonhong — APPROVED** [review]
++ works perfect!

**Comment by iefremov** [comment]
I wonder, if something conflicts with our own accelerator service, maybe we should fix the service instead? I'm not sure that implementing our own accelerator machinery that breaks existing stuff is the right way to go

**Comment by sangwoo108** [comment]
> I wonder, if something conflicts with our own accelerator service, maybe we should fix the service instead? I'm not sure that implementing our own accelerator machinery that breaks existing stuff is the right way to go

I totally agree that we should find out why `Control + A` was added by our service. But the problem is that as our service writes the shortcuts on the Preference, it will persist even if we find and fix the root cause. So in order to help them, we need this patch, I belive.

**Comment by fallaciousreasoning** [comment]
I had a bit more of a dig through `components/remote_cocoa/app_shim/bridged_content_view.mm` and it looks like most of the shortcuts there send `Control` instead of `Command`.

I think the actual bug here is that `Control + A` is being added to the `IDC_SELECT_NEXT_TAB` command when it shouldn't be (looking at the code I can't see what it's being added by).

The `AcceleratorService` has logic for removing default shortcuts from the pref if they've been removed in code, so I think if we can prevent the `Control + A` shortcut being added to `IDC_SELECT_NEXT_TAB` in `brave/browser/ui/views/commands/default_accelerators_mac.h` this will be fixed automatically (and we won't need to write any migrationy code).

**Comment by sangwoo108** [comment]
@fallaciousreasoning Good point! This is what I've got. Looks like Control + Tab in global menu is converted to the Control A.


```
"34016":["Control+KeyA","Command+Shift+BracketRight","Control+PageDown","Command+Alt+ArrowRight"]
```

```
"34016":["Control+KeyA","Command+Shift+BracketRight","Control+PageDown","Command+Alt+ArrowRight"]
```

File: browser/ui/views/commands/default_accelerators_mac.mm
Comment (sangwoo108) [inline browser/ui/views/commands/default_accelerators_mac.mm:72]: Passing 0 to this caused the converting routine to fall back to 'A' - https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:ui/events/keycodes/keyboard_code_conversion_mac.mm;l=653;drc=34b556527508d598afa5fe91626d491904aeaaf2

**Comment by sangwoo108** [comment]
Okay, Updated, Now I've got this with new commit. As @fallaciousreasoning expected, the wrong shortcut seems to be pruned by our service.

```
34016":["Command+Shift+BracketRight","Control+PageDown","Command+Alt+ArrowRight","Control+Tab"]
```

File: browser/ui/views/commands/default_accelerators_mac.mm
Comment (simonhong) [inline browser/ui/views/commands/default_accelerators_mac.mm:58]:
```suggestion
  if (keyCode == '\t') {
```
Curious that `\t` is the only case that we should do like above?

File: browser/ui/views/commands/default_accelerators_mac.mm
Comment (simonhong) [inline browser/ui/views/commands/default_accelerators_mac.mm:59]: ascii code of `\t` is `9`. and `9` is used as `v`'s keycode :)

File: browser/ui/views/commands/default_accelerators_mac.mm
Comment (simonhong) [reply browser/ui/views/commands/default_accelerators_mac.mm:58]: Hmm, confusing.. first charactor could be something like `B` or `D`. and then `keyCode` var stores its ascii value.
In the `B` case, `keyCode` stores `66` and it's passed as `keyCode` params for creating `NSEvent`.
Does `keyCode` of `NSEvent` ctor accept ascii value?

Comment (simonhong) [reply browser/ui/views/commands/default_accelerators_mac.mm:58]: It seems `[keyEquivalent characterAtIndex:0]` is keyboard code. and it should be converted to keycode?

Comment (sangwoo108) [reply browser/ui/views/commands/default_accelerators_mac.mm:58]: Yeah, I think we should intialize keCode with -1.

Comment (simonhong) [reply browser/ui/views/commands/default_accelerators_mac.mm:58]: How about below? 
```
  const unsigned short charCode = [keyEquivalent characterAtIndex:0];
  const int keyCode =
      ui::MacKeyCodeForWindowsKeyCode(static_cast<ui::KeyboardCode>(charCode), /*flags=*/0,
                                      /*us_keyboard_shifted_character=*/nullptr,
                                      /*keyboard_character=*/nullptr);

  NSEvent* keyEvent = [NSEvent keyEventWithType:NSEventTypeKeyDown
                                       location:NSZeroPoint
                                  modifierFlags:0
                                      timestamp:0

...
```

Comment (sangwoo108) [reply browser/ui/views/commands/default_accelerators_mac.mm:58]: Yeah! that works great! Thanks!

File: browser/ui/views/commands/default_accelerators_mac.mm
Comment (sangwoo108) [reply browser/ui/views/commands/default_accelerators_mac.mm]: Nice catch 👍 was using char code not keycode.

**Review by iefremov — APPROVED** [review]

**Comment by bsclifton** [comment]
@sangwoo108 ready to merge? I think this may need an uplift too, right?

**Comment by bsclifton** [comment]
cc: @fallaciousreasoning

**Review by fallaciousreasoning — APPROVED** [review]
Wow, good find @sangwoo108! Thanks so much for working this out.

Based on those prefs, the shortcut should be automatically removed with your update :smile:

**Comment by sangwoo108** [comment]
> @sangwoo108 ready to merge? I think this may need an uplift too, right?

Yes, should be. I was waiting for @fallaciousreasoning 's approval. I created an uplifting https://github.com/brave/brave-core/pull/18956 now.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 13.4 x64 Ventura` using the following build(s):

```
Brave | 1.55.2 Chromium: 115.0.5790.32 (Official Build) nightly (x86_64)
--- | ---
Revision | f9b78cf410ecd5b372310ef25a12ef456c77d90f-refs/branch-heads/5790@{#724}
OS | macOS Version 13.4 (Build 22F66)
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/31059#issue-1757441899, reproduced the original issue using `1.54.69 Chromium: 114.0.5735.133`. Pretting `CMD + A` or `Edit` -> `Select All` would switch to the next tab over via the omnibox rather than selecting the entire text.

Verified that the issue has been resolved using the same STR/Cases mentioned above via `1.55.2 Chromium: 115.0.5790.32` as per the following:

* ensured that pressing `CMD + A` selects the text within the omnibox without issues
* ensured that going through `Edit` -> `Select All` correctly selects the text within the omnibox

**Pull request merged**
